src/main/Main.ts

```diff
@@ -1,12 +1,52 @@
-import {app, ipcMain, nativeImage, BrowserWindow, screen} from "electron";
+import {app, nativeImage, BrowserWindow, screen, globalShortcut, Menu} from "electron";
 import {readFileSync} from "fs";
 import {windowBoundsFilePath} from "../utils/Common";
+import {buildMenuTemplate} from "./Menu";
+
+/**
+ * Contains references to all currently open windows.
+ *
+ * @type {Array}
+ */
+let windows: Electron.BrowserWindow[] = [];
```

Comment:
Do you think it might be possible to have a `Set` of `unfocussedWindows` and a single, optional, `focussedWindow`, with windows moving into and out of the `focussedWindow`? That feels like it might be a little cleaner than an array with an index. I think it would probably also be fine as is if you don't have the time to change it.